### PR TITLE
Enable server delivered pages

### PR DESCRIPTION
**Pull request by JonForest**
This PR enables the collapsing table to work from a external data source providing just the contents of the current page, rather than all the data up front.

It can be extended to handle server-side ordering as well.

### CODE REVIEW

**Review by ewen — APPROVED** [review]
Couldn't follow how `rows` gets set but assume the `updateData` function is responsible for that. But overall LGTM

File: src/components/Table.js [hunk 1]
```diff
@@ -113,15 +113,28 @@ export class Table extends Component {
...
-        });
+        if (!this.props.updateData) {
+            this.setState(currentState => {
+                return nextPage({ state: currentState })
+            });
+            return 
+        }
+
+        const { currentPage, totalPages } = this.state.pagination;
+        if (currentPage < totalPages) this.props.updateData({page: currentPage + 1});
```
Comment (wtcarter) [inline src/components/Table.js:124]: What updates currentPage after nextPage?

**Review by wtcarter — CHANGES_REQUESTED** [review]
Just a question and suggest adding updateData prop to readme

File: src/components/Table.js (hunk 1 above)
Comment (JonForest) [reply src/components/Table.js:124]: This calls a function `updateData` that will send a request to the server, and in the server response the `current_page` is sent, and then comes back into here via updated props.

**Review by wtcarter — APPROVED** [review]

**Pull request merged**